bpf/bpf_host.c

```diff
@@ -76,7 +76,7 @@ static __always_inline int rewrite_dmac_to_host(struct __ctx_buff *ctx,
 	/* Rewrite to destination MAC of cilium_net (remote peer) */
 	if (eth_store_daddr(ctx, (__u8 *) &cilium_net_mac.addr, 0) < 0)
 		return send_drop_notify_error(ctx, src_identity, DROP_WRITE_ERROR,
-					      CTX_ACT_OK, METRIC_INGRESS);
```

Comment:
I'm not against that change but just want to note things are a bit more subtle than they may appear:
Ingress or egress is always with respect to something. In the case of host policy enforcement, it's easy: we're talking of egress/ingress out of/into the host-network namespace. Same for pod policies with pod-network namespaces. But for a packet traveling from pod-to-pod through `cilium_host` with host firewall disabled, should it be considered ingress or egress? :grin:

Reply:
Egh. Yes, I see what you mean. Feels like we're right in the middle of the packet's journey here. And given that it's an unexpected error, probably not worth changing.

Reply:
> In the case of host policy enforcement, it's easy: we're talking of egress/ingress out of/into the host-network namespace.

This just made me have another look at the `from-host` path. If `ipv4_host_policy_egress()` returns an error right now, `tail_handle_ipv4()` reports the drop with `METRIC_INGRESS`. While we would presumably prefer `METRIC_EGRESS`, to match the policy verdict.

Feels way too entangled :/.